lightning/src/util/wakers.rs

```diff
@@ -88,9 +88,16 @@ impl Notifier {
 	/// Wake waiters, tracking that wake needs to occur even if there are currently no waiters.
 	pub(crate) fn notify(&self) {
 		let mut lock = self.notify_pending.lock().unwrap();
-		lock.0 = true;
+		let mut future_probably_generated_calls = false;
 		if let Some(future_state) = lock.1.take() {
-			future_state.lock().unwrap().complete();
+			future_probably_generated_calls |= future_state.lock().unwrap().complete();
+			future_probably_generated_calls |= Arc::strong_count(&future_state) > 1;
+		}
+		if !future_probably_generated_calls {
+			// If a future made some callbacks or still exists (i.e. the state has more than the
+			// one reference we hold), assume the user was notified and skip the "classic"
+			// waiters.
```

Comment:
Not sure I follow what is meant here. (1) The comment seems to be describing `when future_probably_generated_calls` is true but the code is for when it is false. (2) What is mean by "classic"? (3) How does skipping occur?